prusti-specs/src/extern_spec_rewriter.rs

```diff
@@ -226,18 +228,51 @@ pub fn generate_new_struct(item: &syn::ItemImpl) -> syn::Result<syn::ItemStruct>
     new_struct.generics = item.generics.clone();
     let generics = &new_struct.generics;
 
-    let mut fields_str: String = String::new();
+    let mut fields = CommaPunct::<syn::Field>::new();
 
     // Add `PhantomData` markers for each type parameter to silence errors
     // about unused type parameters.
     for param in generics.params.iter() {
-        let field = format!("core::marker::PhantomData<{}>,", param.to_token_stream().to_string());
-        fields_str.push_str(&field);
+        let field: ParsableUnnamedField = match param {
+            syn::GenericParam::Type(tp) => {
+                let ident = tp.ident.clone();
+                syn::parse_quote!{ core::marker::PhantomData<#ident> }
+            }
+            syn::GenericParam::Lifetime(ld) => {
+                let ident = ld.lifetime.clone();
+                syn::parse_quote!{ &#ident core::marker::PhantomData<()> }
+            }
+            syn::GenericParam::Const(_cp) => continue,
+        };
+        // Unwrap the `ParsableUnnamedField` and push. See below.
+        fields.push(field.0);
     }
 
-    let fields : syn::FieldsUnnamed =
-        syn::parse_str(&format!("({})", fields_str)).unwrap();
-
-    new_struct.fields = syn::Fields::Unnamed(fields);
+    new_struct.fields = syn::Fields::Unnamed( syn::parse_quote!{ (#fields) } );
```

Comment:
Does it work if you define `fields` as a `Vec<syn::Field>` and then:
```suggestion
    new_struct.fields = syn::Fields::Unnamed( syn::parse_quote!{ (#(#fields),*) } );
```
If so, you can get rid of the `CommaPunct` bit, and just have `let mut fields = vec![];`

Reply:
Hmm do you think it's more clear like you suggest? I could just import `syn::punctuated::Punctuated` and use `Punctuated<syn::Field, syn::Token![,]>` as the type of `fields` - that way we also do not need the type alias, but avoid this `parse_quote` black magic?

Reply:
Sure, both save space a bit. I think this type of interpolation is used in other places already, so it wouldn't be too out of place :)